src/components/MaanedsbeloepAvansertBeregning/__tests__/hooks.test.tsx

```diff
@@ -0,0 +1,277 @@
+import { IntlProvider } from 'react-intl'
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+
+import { getSelectedLanguage } from '@/context/LanguageProvider/utils'
+// Import the mocked functions for controlling their behavior
+import { useAppSelector } from '@/state/hooks'
+// Import actual selectors to reference them in our mocks
+import {
+  selectCurrentSimulation,
+  selectFoedselsdato,
+} from '@/state/userInput/selectors'
+import { renderHook } from '@/test-utils'
+import { transformUttaksalderToDate } from '@/utils/alder'
+
+import translations_nb from '../../../translations/nb'
+import {
+  hentSumOffentligTjenestepensjonVedUttak,
+  hentSumPensjonsavtalerVedUttak,
+} from '../../Pensjonsavtaler/utils'
+import { usePensjonBeregninger } from '../hooks'
+
+// Mock the hooks that usePensjonBeregninger depends on
+vi.mock('@/state/hooks', () => ({
+  useAppSelector: vi.fn(),
+}))
+
+vi.mock('@/utils/alder', () => ({
+  transformUttaksalderToDate: vi.fn(),
+}))
+
+vi.mock('@/context/LanguageProvider/utils', () => ({
+  getSelectedLanguage: vi.fn(),
+}))
+
+vi.mock('../../Pensjonsavtaler/utils', () => ({
+  hentSumPensjonsavtalerVedUttak: vi.fn(),
+  hentSumOffentligTjenestepensjonVedUttak: vi.fn(),
+}))
+
+const wrapper = ({ children }: { children: React.ReactNode }) => {
+  return (
+    <IntlProvider locale="nb" messages={translations_nb}>
+      {children}
+    </IntlProvider>
+  )
+}
+
+describe('usePensjonBeregninger', () => {
+  // Default mock values to be used across tests
+  const mockUttaksalder: Alder = { aar: 67, maaneder: 0 }
+  const mockGradertUttaksperiode = {
+    uttaksalder: { aar: 62, maaneder: 6 } as Alder,
```

Comment:
Unngå `as` når mulig
```suggestion
    uttaksalder: { aar: 62, maaneder: 6 },
```